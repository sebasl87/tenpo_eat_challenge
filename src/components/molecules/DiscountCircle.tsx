import React from 'react'
import styled from 'styled-components/native';

export interface IDiscountCircle {
    discount?: string;
}

const Circle = styled.View`
width: 30px;
height: 30px;
background: #00BAA4;
justify-content: center;
align-items: center;
position: absolute;
margin-top: 2px;
margin-left: 78px;
z-index: 2;
`
const TextDiscount = styled.Text`
color: #fff;
font-size: 11px;
`
const Discount = styled.Text`
color: #fff;
font-size: 7px;
margin-top: -2px;
`

const DiscountCircle: React.FC<IDiscountCircle> = ({ discount }) => {
    return (
        <Circle style={{ borderRadius: 50 }}>
            <TextDiscount style={{ fontFamily: 'Roboto-Light', }}>{discount}%</TextDiscount>
            <Discount style={{ fontFamily: 'Roboto-Light', }}>DCTO</Discount>
        </Circle>
    )
}

export default DiscountCircle;