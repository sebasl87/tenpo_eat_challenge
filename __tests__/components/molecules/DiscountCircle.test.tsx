import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import DiscountCircle from "../../../src/components/molecules/DiscountCircle";


describe("DiscountCircle component", () => {
    it("should render successfully with DiscountCircle", () => {
        const discountCircle = renderer
            .create(<DiscountCircle discount="DiscountCircle_mock" />)
            .toJSON();

        expect(discountCircle).toMatchSnapshot();
    });
});
