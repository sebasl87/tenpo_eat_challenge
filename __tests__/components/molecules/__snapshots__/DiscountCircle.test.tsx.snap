// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiscountCircle component should render successfully with DiscountCircle 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#00BAA4",
        "height": 30,
        "justifyContent": "center",
        "marginLeft": 78,
        "marginTop": 2,
        "position": "absolute",
        "width": 30,
        "zIndex": 2,
      },
      {
        "borderRadius": 50,
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#fff",
          "fontSize": 11,
        },
        {
          "fontFamily": "Roboto-Light",
        },
      ]
    }
  >
    DiscountCircle_mock
    %
  </Text>
  <Text
    style={
      [
        {
          "color": "#fff",
          "fontSize": 7,
          "marginTop": -2,
        },
        {
          "fontFamily": "Roboto-Light",
        },
      ]
    }
  >
    DCTO
  </Text>
</View>
`;
